war/pom.xml

```diff
@@ -532,42 +570,217 @@ THE SOFTWARE.
                         </execution>
+                    </executions>
+                </plugin>
+                <plugin>
+                    <groupId>com.github.eirslett</groupId>
+                    <artifactId>frontend-maven-plugin</artifactId>
+                    <version>0.0.23</version>
+
+                    <executions>
+
                         <execution>
-                            <id>grunt</id>
-                            <phase>prepare-package</phase>
+                            <phase>initialize</phase>
+                            <id>install node and npm</id>
                             <goals>
-                                <goal>exec</goal>
+                                <goal>install-node-and-npm</goal>
                             </goals>
                             <configuration>
-                                <executable>grunt</executable>
+                                <nodeVersion>v${node.version}</nodeVersion>
+                                <npmVersion>${npm.version}</npmVersion>
+                                <!-- Use the pre-download node and npm packages. See download-maven-plugin config above. -->
+                                <downloadRoot>${project.baseUri}target/frontend/</downloadRoot>
                             </configuration>
                         </execution>
+
+                        <execution>
+                            <phase>initialize</phase>
+                            <id>npm install</id>
+                            <goals>
+                                <goal>npm</goal>
+                            </goals>
+                            <configuration>
+                                <!-- Note that this may not be omitted lest maven-release-plugin be confused (frontend-maven-plugin #109): -->
+                                <arguments>install</arguments>
+                            </configuration>
+                        </execution>
+
+                        <execution>
+                            <phase>generate-sources</phase>
+                            <id>gulp bundle</id>
+                            <goals>
+                                <goal>gulp</goal>
+                            </goals>
+                            <configuration>
+                                <arguments>bundle</arguments>
+                            </configuration>
+                        </execution>
+
+                        <execution>
+                            <phase>test</phase>
+                            <id>gulp test</id>
+                            <goals>
+                                <goal>gulp</goal>
+                            </goals>
+                            <configuration><arguments>bundle test</arguments></configuration>
+                        </execution>
+
                     </executions>
                 </plugin>
             </plugins>
         </build>
     </profile>
-  </profiles>
+
+    <profile>
```

Comment:
why is this in a profile rather than configuring the plugin in pluginManagement so that `mvn clean` just cleans?